src/hooks/useHelpScout.js

```diff
@@ -0,0 +1,35 @@
+import React, { useContext, useState } from 'react'
+import PropTypes from 'prop-types'
+
+const HELPSCOUT_OPTOUT_KEY = 'HELPSCOUT_OPTOUT'
+
+const HelpScoutContext = React.createContext()
+
+function HelpScoutProvider({ children }) {
+  const [optedOut, setOptedOut] = useState(
+    localStorage.getItem(HELPSCOUT_OPTOUT_KEY) === 'true'
+  )
+  const handleOptOut = out => {
+    if (out && window.Beacon) {
+      window.Beacon('destroy')
+    }
+    localStorage.setItem(HELPSCOUT_OPTOUT_KEY, out ? 'true' : 'false')
+    setOptedOut(out)
+  }
```

Comment:
What do you think about making this a `useEffect()` instead?

Something like

```js
useEffect(() => {
  if (optedOut && window.Beacon) {
    window.Beacon('destroy')
  }
  localStorage.setItem(HELPSCOUT_OPTOUT_KEY, out ? 'true' : 'false')
}, [optedOut])
```

That way we could pass `setOptedOut` into the context provider as-is, so that users can expect to use it as a `setState()` hook (i.e. also allow them to pass a function into `setOptedOut`).

Reply:
Yeah, I really like this.